### PR TITLE
Cement perf

### PR DESCRIPTION
**Pull request by stalenestas**
Better cement performance
Visible casing shoes, albeit under the cement still
Bugfix for missing casing end MD in draw
#221

### CODE REVIEW

File: .storybook/src/api/index.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import * as apiReal from './api';
 import * as apiMock from './apiMock';
 
-export * from './api';
+// export * from './api';
 // Un comment if developing or running with local data
-// export * from './apiMock';
+export * from './apiMock';
```
Comment (Brynjulf) [inline .storybook/src/api/index.ts:6]: should change back to using real api

**Comment by stalenestas** [comment]
Added back super simple mock data and fixed the WBP a bit

File: src/layers/CementLayer.ts [hunk 2]
```diff
@@ -72,9 +84,14 @@ export class CementLayer extends WellboreBaseComponentLayer {
         }
       }
 
-      // Always add last point
+      // Always add last points
+      for (let i = c.boc - morePointsForStartAndEndMeters; i < c.boc; i += this.options.wellboreBaseComponentIncrement) {
+        const p = this.referenceSystem.project(i);
+        points.push({ point: new Point(p[0], p[1]), md: i });
+      }
```
Comment (Brynjulf) [inline src/layers/CementLayer.ts]: should validate that `i` is not negative or clamp the value when passing it into list of points.

File: src/layers/CementLayer.ts [hunk 3]
```diff
@@ -60,9 +60,21 @@ export class CementLayer extends WellboreBaseComponentLayer {
...
 
       // Add distance to points
-      for (let i = c.toc; i < c.boc; i += this.options.wellboreBaseComponentIncrement) {
+      for (
+        let i = c.toc + morePointsForStartAndEndMeters;
+        i < c.boc - morePointsForStartAndEndMeters;
+        i += this.options.wellboreBaseComponentIncrement
+      ) {
         const p = this.referenceSystem.project(i);
```
Comment (Brynjulf) [inline src/layers/CementLayer.ts:78]: should validate that `i` is not negative or clamp the value when passing it into list of points.

File: src/layers/CementLayer.ts [hunk 4]
```diff
@@ -60,9 +60,21 @@ export class CementLayer extends WellboreBaseComponentLayer {
       const points = [];
       let prevAngle = 10000;
       const allowedAngleDiff = 0.0005;
+      const morePointsForStartAndEndMeters = 10;
+
+      // Always add first points
+
+      for (let i = c.toc; i < c.toc + morePointsForStartAndEndMeters; i += this.options.wellboreBaseComponentIncrement) {
+        const p = this.referenceSystem.project(i);
+        points.push({ point: new Point(p[0], p[1]), md: i });
```
Comment (Brynjulf) [inline src/layers/CementLayer.ts:69]: should validate that `i` is not negative or clamp the value when passing it into list of points.

**Comment by andreabrambilla** [comment]
Just a general comment. Before merging this, could you please rearrange the commits into meaningful blocks? Eg, one commit about the shoes, one commit about the bugfix, and one commit about the performances?

**Pull request closed without merging**